static/app/views/performance/browser/resources/index.spec.tsx

```diff
@@ -0,0 +1,192 @@
+import {Organization} from 'sentry-fixture/organization';
+
+import {render, screen, waitForElementToBeRemoved} from 'sentry-test/reactTestingLibrary';
+
+import {DetailedOrganization} from 'sentry/types';
+import {useLocation} from 'sentry/utils/useLocation';
+import useOrganization from 'sentry/utils/useOrganization';
+import usePageFilters from 'sentry/utils/usePageFilters';
+import ResourcesLandingPage from 'sentry/views/performance/browser/resources';
+import {SpanFunction, SpanMetricsField} from 'sentry/views/starfish/types';
+
+const {
+  SPAN_SELF_TIME,
+  SPAN_GROUP,
+  HTTP_RESPONSE_CONTENT_LENGTH,
+  SPAN_DOMAIN,
+  SPAN_DESCRIPTION,
+  PROJECT_ID,
+  RESOURCE_RENDER_BLOCKING_STATUS,
+  SPAN_OP,
+} = SpanMetricsField;
+const {SPM, TIME_SPENT_PERCENTAGE} = SpanFunction;
+
+jest.mock('sentry/utils/useLocation');
+jest.mock('sentry/utils/usePageFilters');
+jest.mock('sentry/utils/useOrganization');
+
+const requestMocks: Record<string, jest.Mock> = {};
+
+describe('ResourcesLandingPage', function () {
+  const organization = Organization({
+    features: [
+      'starfish-browser-resource-module-ui',
+      'starfish-view',
+      'performance-database-view',
+    ],
+  });
+
+  beforeEach(() => {
+    setupMocks(organization);
+    setupMockRequests(organization);
+  });
+
+  afterEach(function () {
+    jest.resetAllMocks();
+  });
+
+  it('renders a list of resources', async () => {
+    render(<ResourcesLandingPage />);
+    await waitForElementToBeRemoved(() => screen.queryAllByTestId('loading-indicator'));
+
+    expect(
+      screen.getByRole('cell', {name: 'https://*.sentry-cdn.com/123.js'})
+    ).toBeInTheDocument();
+
+    expect(
+      screen.getByRole('cell', {name: 'https://*.sentry-cdn.com/456.js'})
+    ).toBeInTheDocument();
+  });
+
+  it('contains correct query in charts', async () => {
+    render(<ResourcesLandingPage />);
+    await waitForElementToBeRemoved(() => screen.queryAllByTestId('loading-indicator'));
+
+    expect(requestMocks.mainTable).toHaveBeenCalledWith(
+      expect.anything(),
+      expect.objectContaining({
+        query: expect.objectContaining({
+          query:
+            '!span.description:"browser-extension://*" !resource.render_blocking_status:blocking AND ( span.op:resource.script OR file_extension:css OR file_extension:[woff,woff2,ttf,otf,eot] ) ',
```

Comment:
I find it non-ideal to assert on a string like this, it seems quite brittle because the order of the query conditions often doesn't matter. It won't cause any issues tho, so should be fine.

Reply:
It would be nice to do something like
```
query.hasConditions([!span.description:"browser-extension://*", '!resource.render_blocking_status:blocking', etc
```

Reply:
Instead testing for what is in the query, I'd rather assert for what is being display on the page - we can then even table test with different queries to make sure table still has values.

Reply:
I do like the idea of asserting whats on the page, but my goal here was to find any regressions against queries made in the page. For example, if we mess something up when adding a new file extension. We aren't making real requests here, were mocking responses, so if we can ensure the query doesn't regress, then we can ensure the response data doesn't change (or at least it wouldn't be a FE issue).

Is that something possible to capture with your approach?

Reply:
Without a proper e2e test my solution wouldn't help with that, you're right that this is the best way.

Could we do an inline snapshot test instead then? If all we are looking for is just validating changes in the query?

Reply:
:(
<img width="675" alt="image" src="https://github.com/getsentry/sentry/assets/44422760/6ff3147f-8601-4620-a1b7-947f684251d6">

Reply:
It seems like we can still use them, but we have to manually write the snapshot in, instead of allows jest to write it into code for you. Which is a bit of a bummer

Reply:
I feel like this workaround might be fine (seperate PR/issue, just wanted to discuss)

https://stackoverflow.com/questions/76818626/how-can-one-test-inline-snapshots-with-jest-and-prettier-3-0-0
We don't really need prettier formatting when jest writes in the snapshot, as our CI handles prettier anyways